src/main/java/org/candlepin/subscriptions/tally/billing/BillableUsageController.java

```diff
@@ -136,9 +136,18 @@ private BillableUsage produceMonthlyBillable(BillableUsage usage) {
         usage.getSnapshotDate());
     log.debug("Usage: {}", usage);
 
+    // For old messages, the total value for period is not set yet, so we need to keep using the
+    // getCurrentlyMeasuredTotal method
+    // After SWATCH-2368, we will fully use the provided total value for period from the usage and
+    // the getCurrentlyMeasuredTotal method is to be removed.
```

Comment:
@Sgitario this is a good idea but in this case I think its safe to remove the method here. The fromTallySummary mapping will happen on the same deployment/container/thread without any kafka messaging between when we map the tallySummary and when this method is called. So any new deployment will populate the new field in the mapper and the old ones will still be able to calculate it.

Reply:
@kflahert you were right. My first solution went into the wrong direction since the new field had to be added to the tally snapshot measurement model, not only in the billable usage model. 
I kept this logic here for the same reason I stated in the java comment, but moved the SQL query to the tally snapshot mapper instead. 
PR updated.